runsc/boot/mount_hints.go

```diff
@@ -219,3 +225,60 @@ func (p *PodMountHints) FindMount(mountSrc string) *MountHint {
 	}
 	return nil
 }
+
+// RootfsHint represents extra informations about rootfs that are provided via
+// annotations. They can provide mount source, mount type and overlay information.
+type RootfsHint struct {
+	Mount   specs.Mount `json:"mount"`
+	Overlay string      `json:"overlay"`
+}
+
+func (r *RootfsHint) setType(val string) error {
+	switch val {
+	case erofs.Name, Bind:
+		r.Mount.Type = val
+	default:
+		return fmt.Errorf("invalid type %q", val)
+	}
+	return nil
+}
+
+func (r *RootfsHint) setField(key, val string) error {
+	switch key {
+	case "source":
+		r.Mount.Source = val
+	case "type":
+		return r.setType(val)
+	case "overlay":
+		r.Overlay = val
+	default:
+		return fmt.Errorf("invalid rootfs annotation: %s=%s", key, val)
+	}
+	return nil
+}
+
+// NewRootfsHint instantiates RootfsHint using spec.
+func NewRootfsHint(spec *specs.Spec) (*RootfsHint, error) {
+	var hint *RootfsHint
+	for k, v := range spec.Annotations {
+		// Look for 'dev.gvisor.spec.rootfs' annotations and parse them.
+		if strings.HasPrefix(k, RootfsPrefix) {
```

Comment:
This was to make the coding style be consistent with the existing code in `NewPodMountHints()`:

https://github.com/google/gvisor/blob/8f9211abca9c0564c8c1b1f81d6b28c469b4aba9/runsc/boot/mount_hints.go#L71-L75

But I agree it looks better to avoid some indentation. I will follow your suggestion.